presto-main/src/main/java/io/prestosql/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -158,27 +162,59 @@ private void validateShowStatsSubquery(ShowStats node, Query query, QuerySpecifi
         }
 
-        private Node rewriteShowStats(ShowStats node, Table table, Constraint constraint)
+        private Node rewriteShowStats(ShowStats node, Table table, List<SelectItem> selectItems, Constraint constraint)
         {
+            List<String> statsColumnNames = buildColumnsNames();
             TableHandle tableHandle = getTableHandle(node, table.getName());
             TableStatistics tableStatistics = metadata.getTableStatistics(session, tableHandle, constraint);
-            List<String> statsColumnNames = buildColumnsNames();
-            List<SelectItem> selectItems = buildSelectItems(statsColumnNames);
             TableMetadata tableMetadata = metadata.getTableMetadata(session, tableHandle);
+
             Map<String, ColumnHandle> columnHandles = metadata.getColumnHandles(session, tableHandle);
-            List<Expression> resultRows = buildStatisticsRows(tableMetadata, columnHandles, tableStatistics);
+            Multimap<String, String> extractNamedResultColumns = extractNamedResultColumns(node, tableMetadata, selectItems);
+            List<Expression> resultRows = buildStatisticsRows(tableMetadata, columnHandles, extractNamedResultColumns, tableStatistics);
 
-            return simpleQuery(selectAll(selectItems),
+            return simpleQuery(selectAll(buildSelectItems(statsColumnNames)),
                     aliased(new Values(resultRows),
                             "table_stats_for_" + table.getName(),
                             statsColumnNames));
         }
 
+        private Multimap<String, String> extractNamedResultColumns(ShowStats node, TableMetadata metadata, List<SelectItem> selectItems)
+        {
+            ImmutableMultimap.Builder<String, String> assignments = ImmutableMultimap.builder();
+
+            for (SelectItem item : selectItems) {
+                if (item instanceof AllColumns) {
+                    for (ColumnMetadata column : metadata.getColumns()) {
+                        if (!column.isHidden()) {
+                            assignments.put(column.getName(), column.getName());
+                        }
+                    }
+                }
+
+                if (item instanceof SingleColumn) {
+                    SingleColumn column = (SingleColumn) item;
+                    Optional<Identifier> alias = column.getAlias();
+                    Expression expression = column.getExpression();
+
+                    // TODO: Better error?
+                    check(expression instanceof Identifier, node, "Only table columns names are supported in SELECT");
+
+                    Identifier id = (Identifier) expression;
+                    if (alias.isPresent()) {
+                        assignments.put(id.getValue(), alias.get().getValue());
+                    }
+                    else if (column.getExpression() instanceof Identifier) {
```

Comment:
this `if` is misleading, because it's known to be Identifier.
and it's concerning because there is no further `else`

- remove `if`
- you can add `verify(column.getExpression() instanceof Identifier)` inside `else` block